src/extractor/edge_based_graph_factory.cpp

```diff
@@ -291,8 +297,47 @@ std::vector<NBGToEBG> EdgeBasedGraphFactory::GenerateEdgeExpandedNodes()
 }
 
 /// Actually it also generates turn data and serializes them...
+std::unordered_map<EdgeID, NodeID>
+EdgeBasedGraphFactory::GenerateEdgeExpandedNodesData(const std::string &node_data_filename) const
+{
+    // parallel_for( blocked_range<int>( 0, n ), avg, auto_partitioner( ) );
+
+    NodeID ebg_node_id = 0;
+    std::unordered_map<EdgeID, NodeID> index;
+    EdgeBasedNodeDataExternalContainer node_data_container;
+    for (auto node_id : util::irange(0u, m_node_based_graph->GetNumberOfNodes()))
+    {
+        for (auto edge_id : m_node_based_graph->GetAdjacentEdgeRange(node_id))
+        {
+            const EdgeData &edge_data = m_node_based_graph->GetEdgeData(edge_id);
+            const bool is_encoded_forwards =
+                m_compressed_edge_container.HasZippedEntryForForwardID(edge_id);
+            const bool is_encoded_backwards =
+                m_compressed_edge_container.HasZippedEntryForReverseID(edge_id);
+            if (is_encoded_forwards || is_encoded_backwards)
+            {
+                auto geometry_id =
+                    is_encoded_forwards
+                        ? m_compressed_edge_container.GetZippedPositionForForwardID(edge_id)
+                        : m_compressed_edge_container.GetZippedPositionForReverseID(edge_id);
+
+                BOOST_ASSERT(index.find(edge_id) == index.end());
+                index.insert({edge_id, ebg_node_id++});
```

Comment:
Instead of introducing an own numbering, we can use the (future) `NodeID` from the search graph here. These IDs are computed [here](https://github.com/Project-OSRM/osrm-backend/blob/master/src/extractor/edge_based_graph_factory.cpp#L206-L236). They can be accessed using `GetEdgeData(edge_id).edge_id`.